pkg/destroy/aws/ec2helpers.go

```diff
@@ -296,6 +298,21 @@ func deleteEC2InternetGateway(ctx context.Context, client *ec2.EC2, id string, l
 	return nil
 }
 
+func deleteEC2CarrierGateway(ctx context.Context, client *ec2.EC2, id string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteCarrierGatewayWithContext(ctx, &ec2.DeleteCarrierGatewayInput{
+		CarrierGatewayId: &id,
+	})
+	if err != nil {
+		if err.(awserr.Error).Code() == "InvalidCarrierGatewayID.NotFound" {
```

Comment:
```suggestion
        var awsErr awserr.Error
		if errors.As(err, &awsErr) && awsErr.Code() == "InvalidCarrierGatewayID.NotFound" {
```
`errors.As` can handle error wrapping whereas casting cannot.

Reply:
I know some other parts of the code use the old way but we should eventually fix those too.

Reply:
ACK. Fixed. Thanks.